bones/recordBone.py

```diff
@@ -0,0 +1,299 @@
+
+			else:
+				try:
+					valuesCache[name].append(self._restoreValueFromDatastore(val))
+				except:
+					raise
+					pass
+		else:
+			valuesCache[name] = None
+
+			if isinstance(val, list) and len(val) > 0:
+				try:
+					valuesCache[name] = self._restoreValueFromDatastore(val[0])
+				except:
+					raise
+					pass
+
+			else:
+				if val:
+					try:
+						valuesCache[name] = self._restoreValueFromDatastore(val)
+					except:
+						raise
+						pass
+				else:
+					valuesCache[name] = None
+
+		return True
+
+	def serialize(self, valuesCache, name, entity):
+		if not valuesCache[name]:
+			entity.set(name, None, False)
+
+			if not self.multiple:
+				for k in entity.keys():
+					if k.startswith("%s." % name):
+						del entity[k]
+
+		else:
+			usingSkel = self._usingSkelCache
+
+			if self.multiple:
+				res = []
+
+				for val in valuesCache[name]:
+					usingSkel.setValuesCache(val)
+					res.append(extjson.dumps(usingSkel.serialize()))
+
+				entity.set(name, res, False)
+
+			else:
+				usingSkel.setValuesCache(valuesCache[name])
+				usingData = usingSkel.serialize()
+
+				entity.set(name, extjson.dumps(usingData), False)
+
+				# Copy attrs of our referenced entity in
+				if self.indexed:
+					for k, v in usingData.items():
+						entity["%s.%s" % (name, k)] = v
+
+		return entity
+
+	def fromClient(self, valuesCache, name, data):
+		valuesCache[name] = []
+		tmpRes = {}
+
+		clientPrefix = "%s." % name
+
+		for k, v in data.items():
+			#print(k, v)
+
+			if k.startswith(clientPrefix) or k == name:
+				if k == name:
+					k = k.replace(name, "", 1)
+
+				else:
+					k = k.replace(clientPrefix, "", 1)
+
+				if "." in k:
+					try:
+						idx, bname = k.split(".", 1)
+						idx = int(idx)
+					except ValueError:
+						idx = 0
+
+						try:
+							bname = k.split(".", 1)
+						except ValueError:
+							# We got some garbage as input; don't try to parse it
+							continue
+
+				else:
+					idx = 0
+					bname = k
+
+				if not bname:
+					continue
+
+				if not idx in tmpRes:
+					tmpRes[idx] = {}
+
+				if bname in tmpRes[idx]:
+					if isinstance(tmpRes[idx][bname], list):
+						tmpRes[idx][bname].append(v)
+					else:
+						tmpRes[idx][bname] = [tmpRes[idx][bname], v]
+				else:
+					tmpRes[idx][bname] = v
+
+		tmpList = [tmpRes[k] for k in sorted(tmpRes.keys())]
+
+		errorDict = {}
+		forceFail = False
+
+		if not tmpList and self.required:
+			return "No value selected!"
+
+		for i, r in enumerate(tmpList[:]):
+			usingSkel = self._usingSkelCache
+			usingSkel.setValuesCache({})
+
+			if not usingSkel.fromClient(r):
+				for k, v in usingSkel.errors.items():
+					errorDict["%s.%d.%s" % (name, i, k)] = v
+					forceFail = True
+
+			tmpList[i] = usingSkel.getValuesCache()
+
+		if self.multiple:
+			cleanList = []
+
+			for item in tmpList:
+				err = self.isInvalid(item)
+				if err:
+					errorDict["%s.%s" % (name, tmpList.index(item))] = err
+				else:
+					cleanList.append(item)
+
+			if not cleanList:
```

Comment:
In a customer project we needed to change this line to:
```python
if not cleanList and self.required:
```
This allow us to call fromClient with an empty dict, wich means we will clean this bone and the bone has no value.